File: frontend/src/services/Resume.ts
import { apiClient } from './apiClient';

export interface ApiResponse<T> {
    success: boolean;
    status: number;
    message: string;
    data: T;
}

export interface ResumeCreateRequest {
    name: string;
    email: string;

    age: number;
    gender?: string;

    phone: string;
    address?: string;

    summary?: string;
    selfIntro?: string;

    career?: string;
    projectIntro?: string;

    skills?: string;
    education?: string;
}

export interface ResumeResponse {
    id: number;

    name: string;
    email: string;

    age: number;
    gender?: string;

    phone: string;
    address?: string;

    summary?: string;
    selfIntro?: string;

    career?: string;
    projectIntro?: string;

    skills?: string;
    education?: string;

    createdAt: string;
    updatedAt: string;
}

export const ResumeApi = {

    /**
     * [Resume] 이력서 생성
     */
    create: async (params: ResumeCreateRequest): Promise<ResumeResponse> => {
        const response = await apiClient.post<ApiResponse<ResumeResponse>>(
            "/api/resumes",
            params
        );
        return response.data.data;
    },

    /**
     * [Resume] 전체 조회
     */
    findAll: async (): Promise<ResumeResponse[]> => {
        const response = await apiClient.get<ApiResponse<ResumeResponse[]>>(
            "/api/resumes"
        );
        return response.data.data;
    },

    /**
     * [Resume] 단일 조회
     */
    findById: async (id: number): Promise<ResumeResponse> => {
        const response = await apiClient.get<ApiResponse<ResumeResponse>>(
            `/api/resumes/${id}`
        );
        return response.data.data;
    },

    /**
     * [Resume] 수정
     */
    update: async (
        id: number,
        params: ResumeCreateRequest
    ): Promise<ResumeResponse> => {
        const response = await apiClient.put<ApiResponse<ResumeResponse>>(
            `/api/resumes/${id}`,
            params
        );
        return response.data.data;
    },

    /**
     * [Resume] 삭제
     */
    delete: async (id: number): Promise<void> => {
        await apiClient.delete(`/api/resumes/${id}`);
    }
};

File: backend/src/main/java/io/github/seong5381/interviewMate/resume/entity/Resume.java
package io.github.seong5381.interviewMate.resume.entity;

import io.github.seong5381.interviewMate.auth.domain.User;
import io.github.seong5381.interviewMate.resume.entity.ResumeFeedback;
import jakarta.persistence.*;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Resume {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @OneToOne(mappedBy = "resume", cascade = CascadeType.REMOVE, orphanRemoval = true)
    private ResumeFeedback resumeFeedback;

    // ======================
    // 📌 기본 정보 (필수)
    // ======================
    @Min(0)
    @Max(120)
    private int age;            // 나이

    private String gender;      // 성별

    @NotBlank
    @Column(nullable = false)
    private String phone;       // 전화번호

    private String address;     // 주소

    // ======================
    // 📌 소개
    // ======================
    private String summary;     // 간략 소개

    @Column(length = 5000)
    private String selfIntro;   // 자기소개서

    // ======================
    // 📌 경력
    // ======================
    private String career;      // 경력 요약 (회사/직무)

    @Column(length = 5000)
    private String projectIntro; // 프로젝트 설명 (프로젝트/성과/업무)

    // ======================
    // 📌 스킬
    // ======================
    @Column(length = 1000)
    private String skills;      // Spring Boot, Java, Python 등

    // ======================
    // 📌 학력
    // ======================
    private String education;   // 학교/전공/학위

    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createdAt;
    @LastModifiedDate
    private LocalDateTime updatedAt;

    public String getResumeText() {
        return selfIntro + "\n" + projectIntro;
    }
}